runtime/fastly/builtins/fetch/fetch.cpp

```diff
@@ -0,0 +1,178 @@
+#include "fetch.h"
+#include "../backend.h"
+#include "../fastly.h"
+#include "../fetch-event.h"
+#include "./headers.h"
+#include "./request-response.h"
+#include "encode.h"
+#include "extension-api.h"
+
+using fastly::backend::Backend;
+using fastly::fastly::Fastly;
+using fastly::fetch::Request;
+
+namespace fastly::fetch {
+
+api::Engine *ENGINE;
```

Comment:
If all the other anonymous declaration of `ENGINE` are meant to point to this, it might be better to move this to a different header, and keep the declaration in one `cpp` file. If it's intentional that they're all different, then perhaps this one should move out of the header.

Reply:
I think if we want to change this extension model, we should address that upstream in StarlingMonkey rather.